File: src/data/api/account/models/iot-certificate.response.ts
import { Expose, Type } from 'class-transformer';
import { GoveeAPIResponse } from '../../govee-api.models';

export class IoTCertificateData {
  @Expose({ name: 'endpoint' })
  brokerUrl!: string;

  @Expose({ name: 'p12' })
  p12Certificate!: string;

  @Expose({ name: 'p12Pass' })
  certificatePassword!: string;
}

export class IoTCertificateResponse extends GoveeAPIResponse {
  @Expose({ name: 'data' })
  @Type(() => IoTCertificateData)
  data!: IoTCertificateData;
}

File: govee-python/src/govee/data/api/account/iot_certificate_response.py
"""IoT certificate response model."""

from __future__ import annotations

from dataclasses import dataclass
from typing import Any, Dict


@dataclass
class IoTCertificateData:
    brokerUrl: str
    p12Certificate: str
    certificatePassword: str

    @staticmethod
    def from_dict(d: Dict[str, Any]) -> "IoTCertificateData":
        return IoTCertificateData(
            brokerUrl=d.get("brokerUrl", d.get("endpoint", d.get("brokerUrl", ""))),
            p12Certificate=d.get("p12", d.get("p12Certificate", "")),
            certificatePassword=d.get("p12Pass", d.get("certificatePassword", "")),
        )


@dataclass
class IoTCertificateResponse:
    iotData: IoTCertificateData

    @staticmethod
    def from_dict(d: Dict[str, Any]) -> "IoTCertificateResponse":
        return IoTCertificateResponse(
            iotData=IoTCertificateData.from_dict(d.get("data", {}))
        )
